File: src/main/java/top/chiloven/lukosbot2/commands/impl/StartCommand.java
package top.chiloven.lukosbot2.commands.impl;

import com.mojang.brigadier.CommandDispatcher;
import org.springframework.stereotype.Service;
import top.chiloven.lukosbot2.commands.IBotCommand;
import top.chiloven.lukosbot2.commands.UsageNode;
import top.chiloven.lukosbot2.core.command.CommandSource;

import static top.chiloven.lukosbot2.util.brigadier.builder.LiteralArgumentBuilder.literal;

@Service
public class StartCommand implements IBotCommand {
    @Override
    public String name() {
        return "start";
    }

    @Override
    public String description() {
        return "";
    }

    @Override
    public UsageNode usage() {
        return UsageNode.root(name())
                .description(description())
                .syntax("（内部命令/隐藏命令）")
                .build();
    }

    @Override
    public void register(CommandDispatcher<CommandSource> dispatcher) {
        dispatcher.register(
                literal(name())
                        .executes(ctx -> {
                            ctx.getSource().reply("欢迎使用 LukosBot2！这是由 @chiloven945 制作的聊天机器人，你可以在 Discord 和 Telegram 上找到他！请输入 /help 以查看可用命令。");
                            return 1;
                        })
        );
    }

    @Override
    public boolean isVisible() {
        return false;
    }
}

File: commands/basic/src/main/java/top/chiloven/lukosbot2/commands/bot/StartCommand.kt
/*
 * Copyright © 2026 Chiloven945
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
package top.chiloven.lukosbot2.commands.bot

import org.springframework.stereotype.Service
import top.chiloven.lukosbot2.commands.IBotCommand
import top.chiloven.lukosbot2.core.command.definition.dsl.botCommand

@Service
class StartCommand : IBotCommand {

    private val commandDefinition = botCommand("start") {
        description = "开始使用 LukosBot2"
        visible = false

        execute {
            source.reply("欢迎使用 LukosBot2！这是由 @chiloven945 制作的聊天机器人，你可以在 Discord 和 Telegram 上找到他！发送 /help 查看可用命令。")
        }

        syntax("开始使用机器人")
    }

    override fun definition() = commandDefinition

}
